### PR TITLE
test(refactor): Improve unit tests #295

### LINKED ISSUE #295: Improve unit tests
- All unit test suites have to be isolated and do not rely on other test suites.
- Delete all files created by unit tests after completion.
- Job unit test has to deploy jobs before it's run.
- All of the jobs in the test suite should be created as part of the test run (ie, in a time stamped folder).
- There should be no reliance on jobs that already exist on the SAS Drive, or in non-timestamped folders (to prevent conflicts).

Once it is done we should remove `--runInBand` mode for jest.

### LINKED ISSUE #303: Write `compile` and `compilebuild` tests
Also refactor `main.js` for to use `command` utility for `compileServices` and `compileBuildServices`

### PR DESCRIPTION
**Pull request by saadjutt01**
## Issue

Closes #303 
Closes #295 

## Intent

Local folders will be created with prefixes of their tests:
> ./test-app-build-*
> ./test-app-cb-*
> ./test-app-cbd-*
> ./test-app-cbd-wit
> ./test-app-create-*
> ./test-app-DB-*
> ./test-app-request-*

AppLoc will always be prefixed with `/Public/app/cli-tests/<optional-text><timestamp>`

On completing single test suite, it's local folders and server folder(`appLoc`) will be deleted.

So, after running all tests it will eventually end-up having empty `cli-tests` on server,(considering if for any reason new folders were not deleted, one can manually delete single parent `cli-tests`)

## Implementation

Added tests for `compile`, `compilebuild` and `compile & build`
Added helper function to remove server folder `helper-server-folder.js`
`--runInBand` removed from jest params
Changed logic for `response` to `response.job` in `execute.js` as per new adapter version.

## Checks

- [x] Code is formatted correctly (`npm run lint:fix`).
- [ ] Any new functionality has been unit tested.
- [x] All unit tests are passing (`npm test`).
- [ ] All CI checks are green.
- [ ] JSDoc comments have been added or updated.
- [ ] Reviewer is assigned.
image

After running all tests through Github Action, `cli-tests` folder on drive is **empty** 
image

### CODE REVIEW

File: test/commands/db.spec.js [hunk 1]
```diff
@@ -1,51 +1,54 @@
 import dotenv from 'dotenv'
 import path from 'path'
+import rimraf from 'rimraf'
```
Comment (YuryShkoda) [inline test/commands/db.spec.js:3]: use `deleteFolder` utility

File: test/commands/job.spec.js [hunk 2]
```diff
@@ -1,60 +1,71 @@
...
 
     dotenv.config()
 
-    const targetNameContext = 'cli-tests-context'
-    const config = {
+    config = {
       serverType: process.env.SERVER_TYPE,
       serverUrl: process.env.SERVER_URL,
-      appLoc: '/Public/app/cli-tests',
+      appLoc: `/Public/app/cli-tests/${timestampAppLoc}`,
```
Comment (YuryShkoda) [inline test/commands/job.spec.js:32]: should be target name with timestamp

**Review by YuryShkoda — CHANGES_REQUESTED** [review]

File: test/commands/build.spec.js [hunk 3]
```diff
@@ -18,7 +18,7 @@ describe('sasjs build', () => {
     `should build with minimal template`,
     async () => {
       const timestamp = generateTimestamp()
-      const parentFolderNameTimeStamped = `test-app-${timestamp}-minimal`
+      const parentFolderNameTimeStamped = `test-app-build-minimal${timestamp}`
```
[the remark this answers is not in the document]
Comment (saadjutt01) [reply test/commands/build.spec.js:21]: All tests have `test-app-` as prefix, needs to be specific for deleting local folders for this test suite

File: test/commands/add.spec.js [hunk 4]
```diff
@@ -1,21 +1,28 @@
 import dotenv from 'dotenv'
 import path from 'path'
+import rimraf from 'rimraf'
```
[the remark this answers is not in the document]
Comment (saadjutt01) [reply test/commands/add.spec.js:3]: `rimraf` completely removed from tests

File: test/commands/cb.spec.js [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+} from '../../src/main'
+import { createFolder } from '../../src/utils/file-utils'
+import { generateTimestamp } from '../../src/utils/utils'
+
+describe('sasjs compile', () => {
+  beforeAll(() => {
+    dotenv.config()
+  })
+
+  describe('test-app-timestamp', () => {
```
[the remark this answers is not in the document]
Comment (saadjutt01) [reply test/commands/cb.spec.js:18]: removed unnecessary `describe` block

**Review by allanbowe — APPROVED** [review]

**Review by YuryShkoda — APPROVED** [review]

**Pull request merged**